platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ConfigurationCacheState.kt

```diff
@@ -451,7 +446,7 @@ class ConfigurationCacheState(
                 writeGradleState(gradle)
                 val projects = collectProjects(state.projects, scheduledWork.scheduledNodes, gradle.serviceOf())
                 writeProjects(gradle, projects)
-                writeRequiredBuildServicesOf(state, buildTreeState)
+                writeRequiredBuildServicesOf(state, buildEventListenersOf(build.gradle))
```

Comment:
More for my education than anything else:

I noticed `buildEventListenersOf()` has a filter to exclude build services for "buildSrc". Is that needed for this case? I guess it was not needed before because the listeners were retrieved by build id? Why don't we need to filter out listeners for other (non-buildSrc) builds?

```
    private
    fun buildEventListenersOf(gradle: GradleInternal) =
        gradle.serviceOf<BuildEventListenerRegistryInternal>()
            .subscriptions
            .filterIsInstance<RegisteredBuildServiceProvider<*, *>>()
            .filter(::isRelevantBuildEventListener)

    private
    fun isRelevantBuildEventListener(provider: RegisteredBuildServiceProvider<*, *>) =
        Path.path(provider.buildIdentifier.buildPath).name != BUILD_SRC
```

Reply:
> I noticed `buildEventListenersOf()` has a filter to exclude build services for "buildSrc". Is that needed for this case? I guess it was not needed before because the listeners were retrieved by build id? Why don't we need to filter out listeners for other (non-buildSrc) builds?

That's a good question, because `buildSrc` is not that different from any other plugin-only build. It might be that later changes to handling of builds without work made this filtering obsolete if not incorrect (as you can now run buildSrc tasks in the root build too).